clang/lib/CodeGen/CGExprScalar.cpp

```diff
@@ -2227,14 +2227,6 @@ Value *ScalarExprEmitter::VisitCastExpr(CastExpr *CE) {
     llvm::Value *V = CE->changesVolatileQualification()
                          ? EmitLoadOfLValue(CE)
                          : Visit(const_cast<Expr *>(E));
-    if (V) {
-      // CK_NoOp can model a pointer qualification conversion, which can remove
-      // an array bound and change the IR type.
-      // FIXME: Once pointee types are removed from IR, remove this.
-      llvm::Type *T = ConvertType(DestTy);
-      if (T != V->getType())
-        V = Builder.CreateBitCast(V, T);
-    }
     return V;
```

Comment:
Could fold this into a direct return CE->...

Reply:
I added a follow-up commit for this. Thank you!